.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,55 @@
+name: CodeQL Analysis
+
+on:
+  push:
+    # ignore dependabot branches on push -> https://github.com/microsoft/binskim/issues/425#issuecomment-893373709
+    branches-ignore:
+      - 'dependabot/**'
+  pull_request:
+  schedule:
+    - cron: '0 8 * * *'
+  workflow_dispatch:
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: windows-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v1
+        with:
+          queries: security-and-quality
+          languages: csharp
+
+      - name: Build solution
+        shell: pwsh
+        run: |
+          $pathToSolution = "ApiVersioning.sln"
+          $buildConfiguration = "Release"
+          $useSharedCompilation = "false"
+          $testProjects = @( Get-ChildItem -Path test/ -Recurse | Where-Object { $_.Extension -match 'shproj' -or $_.Extension -match 'csproj' } | ForEach-Object { $_.FullName } )
```

Comment:
This command can be simplified to just:

```powershell
Get-ChildItem -Path test -Include *.csproj -Recurse | Select-Object FullName
```

A couple of supporting points:

- This _assumes_ (or rather _declares_) it needs to run on Windows, but it doesn't _have_ to be restricted to that (in the future)
- Avoid literal `/` or `\` to ensure cross-platform builds; PowerShell will do the _right_ thing
- There is no need to filter _Shared Projects_ (`*.shproj`) separately because they are automatically included in all C# projects (`*.csproj`). _Shared Projects_ do not compile or have their own output.
- There is no need to wrap the results in an array (`@()`) because the results are already an array. There will always be 2+ items.